### PR TITLE
Add feature vector dataloader

### PR DESCRIPTION
**Pull request by XianyuanLiu**
### Description
Update action_dann_lightn example with a dataloader of feature vector input. This dataloader is used in the EPIC challenge. Changes are summarized below. 
1. Add EPIC100DatasetAccess
2. Add new hyperparameters: CLASS_TYPE, INPUT_TYPE and NUM_SEGMENTS
3. Change ClassNetVideo for multi-class output.

### Status
**Work in progress**

If you've been given access to pykale with role Triage or above, on the right (delete these after selection):

- Select **one most appropriate** label.
- **When** your pull request is **ready** for review, select a reviewer. Use the suggested one if unsure.

If you cannot see an option to select a reviewer/label, that means one maintainer will get notified upon your pull request and then select for you.

### Types of changes

- [ ] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] In-line docstrings updated.
- [ ] [Source for documentation at `docs`](https://github.com/pykale/pykale/tree/main/docs/source) manually updated for new API.

### CODE REVIEW

File: kale/prepdata/video_transform.py [hunk 1]
```diff
@@ -79,6 +79,8 @@ def get_transform(kind, image_modality):
             }
         else:
             raise ValueError("Input modality is not in [rgb, flow, joint]. Current is {}".format(image_modality))
+    elif kind is None:
+        return
```
Comment (sz144) [inline kale/prepdata/video_transform.py:83]: ? return what

**Review by sz144 — CHANGES_REQUESTED** [review]

File: examples/action_dann_lightn/config.py [hunk 2]
```diff
@@ -16,15 +16,17 @@
 # Dataset
 # -----------------------------------------------------------------------------
 _C.DATASET = CN()
-_C.DATASET.ROOT = "I:/Datasets/EgoAction/"  # "/shared/tale2/Shared"
+_C.DATASET.ROOT = "J:/Datasets/EgoAction/"  # "/shared/tale2/Shared"
```
[the remark this answers is not in the document]
Comment (XianyuanLiu) [reply examples/action_dann_lightn/config.py:19]: It may be necessary because the data is stored somewhere else (e.g. /Shared/data), not in pykale.

[the remark this answers is not in the document]
Comment (sz144) [reply examples/action_dann_lightn/config.py]: Is it possible to use public dataset for demonstration?

[the remark this answers is not in the document]
Comment (XianyuanLiu) [reply examples/action_dann_lightn/config.py]: It uses public video datasets (e.g. EPIC KITCHENS), but each is large (>10G). Therefore, users need to download and prepare data following dataset instructions before running this example. Then, of course, they need to change configurations fit for data. 

Reasons that this example does not contain video dataset downloading are as follows:
1. Some datasets provide tools or packages to boost downloading speed, but no need to add these into our pykale.
2. Pykale and data may not be in the same hardware due to the datasets' large-scale.

I may update the readme to describe the dataset format. We may add video dataset downloading in the future when we reduce dataset storage by using some dataset tools (e.g. [Hub](https://github.com/activeloopai/Hub)).

[the remark this answers is not in the document]
Comment (haipinglu) [reply examples/action_dann_lightn/config.py]: I agree with Shuo. PyKale examples should be relatively easy to run. Large-scale data experiment can be created on separate repo under PyKale or your personal repo. We can discuss tomorrow.
